### PR TITLE
 Support choosing Prometheus sync mode between federate and remote

### PR DESCRIPTION
**Pull request by thxCode**
**Problem:**
- Remote reader mode only allow `project-level` Prometheus to share the
metrics from `cluster-level` Prometheus
- Remote reader mode cannot save the namespace-related metrics from
`cluster-level` Prometheus

**Solution:**
Add federation support

**Issue:**
#17390

**Related PR:**
https://github.com/rancher/system-charts/pull/13

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: pkg/controllers/user/monitoring/projectHandler.go [hunk 1]
```diff
@@ -208,6 +208,15 @@ func (ph *projectHandler) deployApp(appName, appTargetNamespace string, appProje
 		appAnswers[mustKey] = mustVal
 	}
 
+	// complete sync target & path
+	if appAnswers["prometheus.sync.mode"] == "federate" {
```
Comment (alena1108) [inline pkg/controllers/user/monitoring/projectHandler.go]: Nit: I feel like we can put some repeatable values like `prometheus.sync.target` in constants.

**Comment by cjellick** [comment]
LGTM

**Comment by cjellick** [comment]
nit can be addressed in a follow up

File: pkg/controllers/user/monitoring/projectHandler.go (hunk 1 above)
Comment (thxCode) [reply pkg/controllers/user/monitoring/projectHandler.go]: Got

**Pull request merged**